### PR TITLE
bugfix: remove DoubleSensGeomDet (ph-2 InnerTracker)

### PR DESCRIPTION
**Pull request by emiglior**
#### PR description:

bugfix: this PR fixes the CMSSW issues https://github.com/cms-sw/cmssw/issues/43097 and https://github.com/cms-sw/cmssw/issues/45792 removing the DoubleSensGeomDet introduced for phase-2 Inner Tracker modules (T33 tracker geometry and following T35, T36, T37, T38 and T39) but actually not used in the code (and unlikely to be used in future).

No changes expected for the output.

#### PR validation:

Tested on:
-  issue #43097
   cmsRun Fireworks/Geometry/python/dumpRecoGeometry_cfg.py  tag=2026 version=D110
   (up to version D113)
- issue #45792 
   cmsRun Alignment/TrackerAlignment/test/Misalignments/createRandomlyMisalignedGeometry_Phase2_cfg.py
- D110: runTheMatrix.py --what upgrade -l 29607.0  
- D112: runTheMatrix.py --what upgrade -l 30407.0  

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport (there will be a companion backport for 14_1_X cycle)

### CODE REVIEW

**Comment by emiglior** [comment]
@henriettepetersen @Raffaella07 @bartosik-hep @sroychow @mdelcourt FYI

**Comment by alja** [comment]
+1

File: Alignment/TrackerAlignment/test/Misalignments/createRandomlyMisalignedGeometry_Phase2_cfg.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import copy, sys, os
 
 import Configuration.Geometry.defaultPhase2ConditionsEra_cff as _settings
-GLOBAL_TAG, ERA = _settings.get_era_and_conditions("2026D107")  # this should really be _settings.DEFAULT_VERSION :(
+GLOBAL_TAG, ERA = _settings.get_era_and_conditions("2026D110")  # this should really be _settings.DEFAULT_VERSION :(
```
Comment (mmusich) [inline Alignment/TrackerAlignment/test/Misalignments/createRandomlyMisalignedGeometry_Phase2_cfg.py:7]:
```suggestion
GLOBAL_TAG, ERA = _settings.get_era_and_conditions(_settings.DEFAULT_VERSION)
```

File: Alignment/TrackerAlignment/test/Misalignments/createRandomlyMisalignedGeometry_Phase2_cfg.py [hunk 2]
```diff
@@ -48,7 +48,7 @@
 ###################################################################
 # Ideal geometry producer and standard includes
 ###################################################################
-process.load('Configuration.Geometry.GeometryExtended2026D107Reco_cff')
+process.load('Configuration.Geometry.GeometryExtended2026D110Reco_cff')
```
Comment (mmusich) [inline Alignment/TrackerAlignment/test/Misalignments/createRandomlyMisalignedGeometry_Phase2_cfg.py:51]:
```suggestion
process.load('Configuration.Geometry.GeometryExtended2026DefaultReco_cff')
```

**Review by mmusich — CHANGES_REQUESTED** [review]
This way the test will be automatically forward-ported the next time we move the default geometry.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
type bug-fix

**Comment by emiglior** [comment]
when testing the branch on lxplus, I had similar failures that were solved with 

setenv FRONTIER_FORCERELOAD softlong

**Comment by mmusich** [comment]
please test

File: Geometry/TrackerGeometryBuilder/test/python/testTrackerHierarchy_cfg.py [hunk 3]
```diff
@@ -1,10 +1,14 @@
 import FWCore.ParameterSet.Config as cms
 
-process = cms.Process("GeometryTest")
+from Configuration.Eras.Era_Phase2C17I13M9_cff import Phase2C17I13M9
```
Comment (mmusich) [inline Geometry/TrackerGeometryBuilder/test/python/testTrackerHierarchy_cfg.py:3]: there's already a file that does this:

https://github.com/cms-sw/cmssw/blob/c312b85e0474fb1c058f8e2c8e0aadcf6b69c30d/Geometry/TrackerGeometryBuilder/test/python/testPhase2TrackerHierarchy_cfg.py#L19-L29

introduced in https://github.com/cms-sw/cmssw/pull/45866. I would revert the changes to this file (and keep it as reference for the Run3 geometry)

Comment (emiglior) [reply Geometry/TrackerGeometryBuilder/test/python/testTrackerHierarchy_cfg.py]: change reverted and branch updated

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+alca

**Comment by jfernan2** [comment]
+1

**Comment by mmusich** [comment]
urgent
   * it would be desirable if the backport PR  https://github.com/cms-sw/cmssw/pull/45996 makes `CMSSW_14_1_0`.

**Comment by alja** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by mandrenguyen** [comment]
+1
